src/transports/firebase.ts

```diff
@@ -0,0 +1,41 @@
+// Third party dependencies
+import { initializeApp } from "firebase/app";
+import { getFirestore, doc, setDoc, Firestore } from "firebase/firestore";
+import dotenv from "dotenv";
+import { Transport } from "./transport";
+
+// Local dependencies
+import { LogEntry } from "../log-entry";
+
+dotenv.config();
+
+const firebaseConfig = {
+  apiKey: process.env.FIREBASE_API_KEY,
+  authDomain: process.env.FIREBASE_AUTH_DOMAIN,
+  projectId: process.env.FIREBASE_PROJECT_ID,
+  storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
+  appId: process.env.FIREBASE_APP_ID,
+  measurementId: process.env.FIREBASE_MEASUREMENT_ID,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+export const db: Firestore = getFirestore(app);
```

Comment:
Move these to the `send` function. I don't expect 'send' to be invoked often but these objects would be initialized with every import as-is.